.github/workflows/e2e.yml

```diff
@@ -14,6 +14,7 @@ jobs:
   main:
     name: Run end-to-end tests
     runs-on: depot-ubuntu-22.04-16
+    timeout-minutes: 25
```

Comment:
can we move this down to the "Run tests" step? that's where it was stalling before:
<img width="1167" alt="image" src="https://github.com/latticexyz/mud/assets/508855/7b52e19a-8257-45ef-8541-c9c5e8590ea6">

maybe also worth lowering this to ~15 mins as it seems the average time on main is ~5 mins for this step